tests/FINDOLOGIC/Tests/Objects/JsonResponseTest.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+namespace FINDOLOGIC\Tests\Objects;
+
+use FINDOLOGIC\Objects\JsonResponse;
+use PHPUnit\Framework\TestCase;
+
+class JsonResponseTest extends TestCase
+{
+    /**
+     * Will use a real response that could come from a request. It returns the Object.
+     *
+     * @param string $filename
+     * @return JsonResponse
+     */
+    public function getRealResponseData($filename = 'demoResponseSuggest.json')
+    {
+        // Get contents from a real response locally.
+        $realResponseData = file_get_contents(__DIR__ . '/../../Mockdata/' . $filename);
+        return new JsonResponse($realResponseData);
+    }
+
+    public function testResponseWillReturnExpectedLabels()
+    {
+        $expectedLabels = [
+            'siemens',
+            'sama',
+            'Schulthess',
+            'Siemens',
+            'Sale%',
+            'Staubsauger',
+            'Miele Dampfgarer mit Mikrowelle DGM 6600, 60 cm, Edelstahl, SensorTronic, MultiSteam, 1000 W',
+            'Bosch Geschirrspüler SPV40E40EU, Vollintegrierbar, 45 cm, 9 Massgedecke, A+',
+            'Bosch Geschirrspüler SBA88TD16E PerfectDry XXL Geschirrspüler 60cm mit Home Connect Vollintegrierbar',
+            'Bosch Geschirrspüler SBE46MX03E SuperSilence XXL Geschirrspüler 60cm Vollintegrierbar mit VarioScharnier',
+        ];
+        $response = $this->getRealResponseData();
+
+        $count = 0;
+        foreach ($response->getSuggestions() as $suggestion) {
```

Comment:
You can use `… as $count => $suggestion)` or maybe ' foreach (array_values($response->getSuggestions()) as $suggestion) }` to get rid of manually handling `$count`.

Or maybe build an array of actual suggestion labels using some `array_map` magic and then comparing the expected and actual label array.

Reply:
> Or maybe build an array of actual suggestion labels using some `array_map` magic and then comparing the expected and actual label array.

Or maybe we are just both thinking too complex. Here is how i fixed it 😆:

```php
foreach ($response->getSuggestions() as $key => $suggestion) {
    $this->assertEquals($expectedLabels[$key], $suggestion->getLabel());
}
```